helpers/apolloLogPlugin.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ApolloServerPlugin,
+  GraphQLRequestListener,
+  GraphQLRequestContext
+} from 'apollo-server-plugin-base'
+
+import * as Sentry from '@sentry/nextjs'
+
+/**
+ * Temporary solution to handle Sentry.flush() possibly rejecting with 'false'.
+ * TODO: Remove when issue is resolved with sentry package
+ * @see https://github.com/getsentry/sentry-javascript/issues/3643
+ * @param timeout — Maximum time in ms the client should wait.
+ */
+const sentryTryFlush: any = async (timeout?: number) => {
+  try {
+    await Sentry.flush(timeout)
+  } catch (error) {
+    console.error('Flush failed: ', error)
+  }
+}
+
+/**
+ * Plugin to capture errors during apollo transaction and send a copy to sentry/logflare
+ */
+export const apolloLogPlugin: ApolloServerPlugin = {
+  requestDidStart<TContext>(
+    _: GraphQLRequestContext<TContext>
+  ): GraphQLRequestListener<TContext> {
+    return {
+      didEncounterErrors(ctx) {
+        for (const err of ctx.errors) {
+          Sentry.withScope(scope => {
+            scope.setTag('kind', ctx.operation?.operation)
+            scope.setExtra('query', ctx.request.query)
+            scope.setExtra('variables', ctx.request.variables)
+
+            if (err.path) {
+              scope.addBreadcrumb({
+                category: 'query-path',
+                message: err.path.join(' > '),
+                level: Sentry.Severity.Debug
+              })
+            }
+
+            const transactionId =
+              ctx.request.http?.headers.get('x-transaction-id')
+            if (transactionId) {
+              scope.setTransactionName(transactionId)
+            }
+            Sentry.captureException(err)
+          })
+          // Log error so logflare can also receive a copy of the error
+          console.error(err)
```

Comment:
Shouldn't we use winston-logger for printing errors?

Reply:
The main reason I opted to just use a simple console.error is because it plays nicely with our vercel hosted serverless setup. The winston setup was using an old sentry transporter that will not work with our vercel hosting, which is why we currently see no logs.

I also ran into issues with trying to get our custom `LoggedRequest` type to work with the new sentry wrapper `withSentry` in the api/lessons route and then also integrating it into the apollo plugin `GraphQLRequestContext` type. 

It's really about getting our logging to play nicely with AWS Lambda functions that are wrapped in some vercel function that isnt that well documented (they just say use our integrations).

I also believe our winston current setup is responsible for this error (which we are currently seeing in production) https://github.com/garageScript/c0d3-app/issues/924 and would like to phase it out completely in another pr if that is the case or at least fix the issue. With our serverless hosting winston just seems to add more headache than it's worth.